nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -286,17 +287,20 @@ sub GrubFs {
     ";
 
     if ($font) {
+	    my $gfxpayloadEfi = $gfxpayload ? $gfxpayload : "keep";
+	    my $gfxpayloadBios = $gfxpayload ? $gfxpayload : "text";
```

Comment:
Is there anything stopping us from setting the defaults within the NixOS module instead of in the generation script? I'm thinking "just like gfxmodeEfi/gfxmodeBios", even though it's a bit inelegant.

It looks like the configuration file generated is expected to be able to be used on both EFI and Legacy boot setups concurrently. I think this assumption should be kept true as much as possible. With this change, it would be impossible to *set* the default values within the NixOS module due to how the `grub_platform` check happens at runtime in the generated config.

I think this sounds reasonable, considering how all other option's default values are described in the module.

Reply:
Ok, so the only way I see it being possible, is to have two top-level properties `gfxpayloadEfi` and `gfxpayloadBios` on `grub`, defaulting to `text` and `keep`, respectively. How about that?

Reply:
Ok, so the only way I see it being possible, is to have two top-level properties gfxpayloadEfi and gfxpayloadBios on grub, defaulting to text and keep, respectively. How about

Reply:
Yes, two nixos options, like it's currently done for `gfxmodeEfi` and `gfxmodeBios`; I presume they would be named `gfxpayloadEfi` and `gfxpayloadBios`.

Reply:
yup, will do tomorrow.

Reply:
tested on both EFI and BIOS. Looking good.